src/framework/handlers/bundle.js

```diff
@@ -41,59 +37,38 @@ class BundleHandler {
             };
         }
 
-        const self = this;
-
-        http.get(url.load, {
-            responseType: Http.ResponseType.ARRAY_BUFFER,
-            retry: this.maxRetries > 0,
-            maxRetries: this.maxRetries
-        }, function (err, response) {
-            if (!err) {
-                try {
-                    self._untar(response, callback);
-                } catch (ex) {
-                    callback('Error loading bundle resource ' + url.original + ': ' + ex);
-                }
-            } else {
-                callback('Error loading bundle resource ' + url.original + ': ' + err);
-            }
-        });
-    }
+        fetch(url.load, {
+            mode: 'cors',
+            credentials: 'include'
+        }).then((res) => {
```

Comment:
Does `fetch` still handle `maxRetries` (I can't find it in diff)?

Reply:
Could probably write something like this to wrap fetch to handle retries and timeout:

```js
const fetchExtra = async (url, { retry = false, maxRetries = 5, timeout = 5000, ...options } = {}) => {
    const timeoutPromise = new Promise((_, reject) => setTimeout(() => reject(new Error('Timed out.')), timeout));
    const fetchPromise = new Promise(async (resolve, reject) => {
        if (!retry) {
            try {
                resolve(await fetch(url, options));
            } catch (e) {
                reject(e);
            }
        }

        let attempts = 0;
        while (attempts < maxRetries) {
            try {
                resolve(await fetch(url, options));
                break;
            } catch (e) {
                attempts++;
                if (attempts === maxRetries) {
                    reject(e);
                    break;
                }
            }
        }
    });

    return await Promise.race([ fetchPromise, timeoutPromise ]);
};

```

Reply:
@Maksims the fetch method still needs to handle max retries